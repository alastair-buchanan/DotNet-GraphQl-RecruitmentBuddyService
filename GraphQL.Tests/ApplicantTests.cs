using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using RecruitmentBuddy.GraphQL.Applicants;
using RecruitmentBuddy.GraphQL.Data;
using RecruitmentBuddy.GraphQL.Types;
using Snapshooter.Xunit;
using Xunit;

namespace GraphQL.Tests;

public class ApplicantTests
{
    [Fact]
        public async Task Applicant_Schema_Changed()
        {
            // arrange
            // act
            ISchema schema = await new ServiceCollection()
                .AddPooledDbContextFactory<ApplicationDbContext>(
                    options => options.UseInMemoryDatabase("Data Source=recruitment.db"))
                .AddGraphQL()
                .AddQueryType(d => d.Name("Query"))
                .AddTypeExtension<ApplicantQueries>()
                .AddMutationType(d => d.Name("Mutation"))
                .AddTypeExtension<ApplicantMutations>()
                .AddType<ApplicantType>()
                .EnableRelaySupport()
                .BuildSchemaAsync();
            
            // assert
            schema.Print().MatchSnapshot();
        }
        
        /*[Fact]
        public async Task RegisterApplicant()
        {
            // arrange
            IRequestExecutor executor = await new ServiceCollection()
                .AddPooledDbContextFactory<ApplicationDbContext>(
                    options => options.UseInMemoryDatabase("Data Source=recruitment.db"))
                .AddGraphQL()
                .AddQueryType(d => d.Name("Query"))
                .AddTypeExtension<ApplicantQueries>()
                .AddMutationType(d => d.Name("Mutation"))
                .AddTypeExtension<ApplicantMutations>()
                .AddType<ApplicantType>()
                .EnableRelaySupport()
                .BuildRequestExecutorAsync();

            // act
            IExecutionResult result = await executor.ExecuteAsync(@"
            mutation RegisterApplicant{
                registerApplicant(
                    input: {
                        emailAddress: ""michael@chillicream.com""
                            firstName: ""michael""
                            lastName: ""staib""
                            userName: ""michael3""
                        })
                {
                    applicant {
                        id
                    }
                }
            }");

            // assert
            result.ToJson().MatchSnapshot();
        }*/
}