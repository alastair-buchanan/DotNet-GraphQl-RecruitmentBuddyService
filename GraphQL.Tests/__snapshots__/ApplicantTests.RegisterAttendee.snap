{
  "errors": [
    {
      "message": "The field \u0060registerApplicant\u0060 does not exist on the type \u0060Mutation\u0060.",
      "locations": [
        {
          "line": 3,
          "column": 17
        }
      ],
      "extensions": {
        "type": "Mutation",
        "field": "registerApplicant",
        "responseName": "registerApplicant",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      }
    }
  ]
}
